controller/src/beerocks/master/son_management.cpp

```diff
@@ -1950,8 +1978,12 @@ void son_management::handle_bml_message(Socket *sd,
         }
         if (channel_pool != nullptr && channel_pool_size != CHANNEL_SCAN_INVALID_PARAM &&
             op_error_code == eChannelScanOperationCode::SUCCESS) {
+            auto u_channel_pool_size = uint8_t(channel_pool_size);
             auto channel_pool_set =
                 std::unordered_set<uint8_t>(channel_pool, channel_pool + channel_pool_size);
+            if (channel_pool_size == 1 && *channel_pool_set.begin() == SCAN_ALL_CHANNELS)
```

Comment:
I understand why @tomereli  doesn't like the `*channel_pool_set.begin() == SCAN_ALL_CHANNELS)`
if we are going to "learn to live with it" - at least wrap it with a meaningful naming function:
```
bool is_scan_all_channels_request(channel_pool_set_type &channel_pool_set){
    return ((channel_pool_set == 1) &&*channel_pool_set.begin() == SCAN_ALL_CHANNELS));
}
```
or something similar

Reply:
The helper function actually makes the code look much better.
Changed